### DISCUSSION TITLE
Enable resultCache by default for all Find queries

### DISCUSSION THREAD

**Discussion by noftaly** [mikro-orm/mikro-orm#4656] [Ideas]
**Is your feature request related to a problem? Please describe.**
I think that the fact that the result cache is opt-in is a bit counter intuitive and discourages its use.
It would be great if we could instead enable it by default on some (or all?) entities, so we wouldn't have to add `{ cache: true }` in all the queries we want to have cache in.

**Describe the solution you'd like**
To enable it for some entities, one way to do that would be through the `@Entity()` decorator, where we could have a `cache?: boolean | number | [string, number];` property to enable it by default. If we want to opt out of the cache for some queries then we could do `{ cache: false }` inside the `.find()` of our query.
Then in the `EntityManager`'s `find*` methods we would use `options.cache` if it's set, and the entity metadata's `.cache` otherwise.

If we want to enable it globally by default, then we could do the same thing in `MikroORM#discoverEntities`, set the entity metadata's cache to what it was set at, and if it is not set then default it to the value defined in the configuration.

**Describe alternatives you've considered**
Adding `{ cache: true }` to all our find queries.

**Comment by B4nan** [comment]
This feels pretty weird, do you know any other tool that would do it? I dont think you should be caching all queries, its the very opposite. Enabling this globally could have pretty weird side effects.

**Reply by noftaly** [reply]
I don't know any tools that do that but I haven't researched much

What side effects are you thinking of?
I have a table that has a lot of reads in bursts, but very few writes. Enabling a result cache of a couple of seconds on it could reduce its stress when there are bursts.
In addition, for my application, even if the result cache was enabled globally for all entities for 1 or 2 seconds, I don't really see what harm that could cause appart from some queries not being up to date by a second (which again, would not be a problem in my application).
If you think this could also impact the memory negatively then we could just have an option `maxEntries` in the `MemoryCacheAdapter` for example.

**Comment by B4nan** [comment]
I guess it won't hurt anyone as it's opt-in...

https://github.com/mikro-orm/mikro-orm/commit/5876c99e398919ca63ce105c644a8621813ffffa

**Reply by noftaly** [reply]
Thank you for working on this!
